### PR TITLE
work_on_booking_seats_work_in_progress

### PR DESCRIPTION
**Pull request by studentrohit123**

### CODE REVIEW

File: app/models/booking.rb [hunk 1]
```diff
@@ -5,6 +5,17 @@ class Booking < ApplicationRecord
...
 
-  validates :name, :age, :gender, :date, :seat_type, presence: true
+  #validates :name, :age, :gender, :date, :seat_type, presence: true
 
+  validate :seat_must_be_available
+
+  private
+
+  def seat_must_be_available
+    byebug
```
Comment (Kapil0187) [inline app/models/booking.rb]: don't push byebug , debugger in github

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -20,6 +22,7 @@ def new
   end
 
   def create
+    byebug
```
Comment (Kapil0187) [inline app/controllers/bookings_controller.rb:25]: remove byebug from here

File: app/views/bookings/new.html.erb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 <h1 class="text-center">New Booking</h1>
+<%= render 'shared/errors', object: @booking %>
+<% byebug %>
```
Comment (Kapil0187) [inline app/views/bookings/new.html.erb:3]: remove byebug from here

**Review by Kapil0187 — CHANGES_REQUESTED** [review]

File: app/models/booking.rb (hunk 1 above)
Comment (studentrohit123) [reply app/models/booking.rb]: ok

File: app/controllers/bookings_controller.rb [hunk 4]
```diff
@@ -50,6 +53,10 @@ def set_booking
     @booking = Booking.find(params[:id])
   end
 
+  def set_bus
```
Comment (vinayaklovanshi) [inline app/controllers/bookings_controller.rb]: Method name is incorrect, why we are taking it as set_bus when we are finding tain?

File: app/views/bookings/new.html.erb (hunk 3 above)
Comment (vinayaklovanshi) [inline app/views/bookings/new.html.erb]: Why you are pushing byebug everywhere?

**Review by vinayaklovanshi — CHANGES_REQUESTED** [review]

File: app/controllers/bookings_controller.rb [hunk 5]
```diff
@@ -4,6 +4,8 @@ class BookingsController < ApplicationController
 
   before_action :set_booking, only: [:show, :edit, :update, :destroy]
```
Comment (vinayaklovanshi) [inline app/controllers/bookings_controller.rb:6]: Remove this extra space

File: app/controllers/bookings_controller.rb [hunk 6]
```diff
@@ -45,16 +47,21 @@ def destroy
...
   end
-
+  
+  private
   def set_booking
     @booking = Booking.find(params[:id])
   end
 
-  private
+  def set_bookable
```
Comment (vinayaklovanshi) [inline app/controllers/bookings_controller.rb:56]: Use this:

@bookable = params[:bookable_type].constantize.find(params[:bookable_id])

**Review by vinayaklovanshi — CHANGES_REQUESTED** [review]

**Pull request merged**